### PR TITLE
[TextFields] Remove MDCTextFieldTypographyThemer from snapshot tests

### PR DESCRIPTION
**Pull request by andrewoverton**
This goal of this PR is to remove usage of MDCTextFieldTypographyThemer from MDCTextField snapshot tests in preparation for its deprecation and deletion. In cases where theming with a theming extension/container scheme is available I'm changing the snapshot test files to use those. In cases where there is no equivalent theming extension I am foregoing theming. The two things that have themers but lack theming extensions are MDCTextField itself (we only have theming extensions for the filled and outlined textfield controllers) and the MDCTextInputControllerOutlinedTextArea.

Related to #9168

### CODE REVIEW

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m [hunk 1]
```diff
@@ -44,14 +43,9 @@ - (void)setUp {
...
   MDCSemanticColorScheme *colorScheme =
       [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
-  MDCTypographyScheme *typographyScheme =
-      [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201804];
 
   [MDCOutlinedTextFieldColorThemer applySemanticColorScheme:colorScheme
                                       toTextInputController:self.textFieldController];
-  [MDCTextFieldTypographyThemer applyTypographyScheme:typographyScheme
-                                toTextInputController:self.textFieldController];
-  [MDCTextFieldTypographyThemer applyTypographyScheme:typographyScheme toTextInput:self.textField];
```
Comment (galiak11) [inline components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:54]: Aren't you missing a call to theme self.textField?
The theming extension seem to applySemanticColorScheme only to self.textFieldController, not to self.textField, and themers seem to apply theming to both.

**Review by galiak11 — APPROVED** [review]
The move from themers to theming extensions changes not only the API, but also the versions of the schemes. I expect that should lead to a lot of scuba changes. However, the changes to size of the scubas make it difficult to see what the visual changes actually are.

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m [hunk 2]
```diff
@@ -44,14 +43,9 @@ - (void)setUp {
   self.textFieldController.characterCountViewMode = UITextFieldViewModeAlways;
   MDCSemanticColorScheme *colorScheme =
       [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
-  MDCTypographyScheme *typographyScheme =
-      [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201804];
 
   [MDCOutlinedTextFieldColorThemer applySemanticColorScheme:colorScheme
```
Comment (romoore) [inline components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:47]: If there is no theming extension, please add a TODO to re-apply the theming once it's created.

**Review by romoore — DISMISSED** [review]

**Review by romoore — COMMENTED** [review]
Please provide a bit more context in the PR summary to explain why (as Galia pointed out) the Theming extension is only on the controller but the Themers were on both the controller and field.

**Comment by andrewoverton** [comment]
@galiak11, I don't think this PR itself will result in any scuba changes because it is just making changes to our test files. Also, nobody is using MDCTextFieldTypographyThemer internally, so nobody will have to migrate to theming extensions from that and experience unexpected changes.

@romoore, I updated the PR description 👍

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m (hunk 1 above)
Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:54]: There is no theming extension for MDCTextField so I'm only using the theming extension on the controllers.

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m (hunk 2 above)
Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:47]: I personally don't think adding theming extensions for MDCTextInputControllerOutlinedTextArea is a worthwhile endeavor. I'm hoping that MDCTextControl text areas become the preferred solution for text areas before that work would be planned. What do you think about that?

Comment (romoore) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:47]: We may decide not to build Theming Extensions for the existing Text Fields, but having a TODO to track the work and know where to update will assist us if we do decide to build them.

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m (hunk 1 above)
Comment (romoore) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:54]: I think the controllers all have access to their textfields via the `.textInput` property.

Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerBaselineCharacterCountArabicSnapshotTests.m:54]: They do, but I didn't want to modify them in this PR.

**Pull request merged**